lib/app/config.ts

```diff
@@ -0,0 +1,175 @@
+import * as props from '../properties.js';
+import type {ConfigLoadOptions, ConfigurationResult} from './config.interfaces.js';
+import {isDevMode, measureEventLoopLag} from './utils.js';
+
+/**
+ * Creates the property hierarchy for configuration loading
+ */
+export function createPropertyHierarchy(env: string[], useLocalProps: boolean): string[] {
+    const propHierarchy = [
+        'defaults',
+        env,
+        env.map(e => `${e}.${process.platform}`),
+        process.platform,
+        os.hostname(),
+    ].flat();
+
+    if (useLocalProps) {
+        propHierarchy.push('local');
+    }
+
+    logger.info(`properties hierarchy: ${propHierarchy.join(', ')}`);
+    return propHierarchy;
+}
+
+/**
+ * Filter languages based on wanted languages from configuration
+ */
+export function filterLanguages(
+    wantedLanguages: string[] | undefined,
+    existingLanguages: Record<LanguageKey, Language>,
+): Record<LanguageKey, Language> {
+    if (wantedLanguages) {
+        const filteredLangs: Partial<Record<LanguageKey, Language>> = {};
+        for (const wantedLang of wantedLanguages) {
+            for (const lang of Object.values(existingLanguages)) {
+                if (lang.id === wantedLang || lang.name === wantedLang || lang.alias.includes(wantedLang)) {
+                    filteredLangs[lang.id] = lang;
+                }
+            }
+        }
+        // Always keep cmake for IDE mode, just in case
+        filteredLangs[existingLanguages.cmake.id] = existingLanguages.cmake;
+        return filteredLangs as Record<LanguageKey, Language>;
+    }
+    return existingLanguages;
+}
+
+/**
+ * Configure event loop lag monitoring
+ */
+export function setupEventLoopLagMonitoring(ceProps: PropertyGetter): void {
+    const lagIntervalMs = ceProps('eventLoopMeasureIntervalMs', 0);
+    const thresWarn = ceProps('eventLoopLagThresholdWarn', 0);
+    const thresErr = ceProps('eventLoopLagThresholdErr', 0);
+
+    let totalLag = 0;
+    const ceLagSecondsTotalGauge = new PromClient.Gauge({
+        name: 'ce_lag_seconds_total',
+        help: 'Total event loop lag since application startup',
+    });
+
+    async function eventLoopLagHandler() {
+        const lagMs = await measureEventLoopLag(lagIntervalMs);
+        totalLag += Math.max(lagMs / 1000, 0);
+        ceLagSecondsTotalGauge.set(totalLag);
+
+        if (thresErr && lagMs >= thresErr) {
+            logger.error(`Event Loop Lag: ${lagMs} ms`);
+        } else if (thresWarn && lagMs >= thresWarn) {
+            logger.warn(`Event Loop Lag: ${lagMs} ms`);
+        }
+
+        setImmediate(eventLoopLagHandler);
+    }
+
+    // Only setup monitoring if interval is set
+    if (lagIntervalMs > 0) {
+        setImmediate(eventLoopLagHandler);
+    }
+}
+
+/**
+ * Load and initialize application configuration
+ */
+export function loadConfiguration(options: ConfigLoadOptions): ConfigurationResult {
+    const {appArgs, options: opts, propDebug} = options;
```

Comment:
Perhaps destructure inline?
```suggestion
export function loadConfiguration({appArgs, options: opts, propDebug}: ConfigLoadOptions): ConfigurationResult {
```